### ISSUE TITLE
Disable CS0660 on ref structs

### ISSUE THREAD

**Issue by juliusfriedman** [dotnet/roslyn#45799]
They cannot implement `Equals(object)`

**Comment by jaredpar** [comment]
A `ref struct` can override `Equals` and explicitly throw as it's not supported (and can't be supported)

```c#
using System;
ref struct S { 
    public static bool operator==(S left, S right) => throw null;
    public static bool operator!=(S left, S right) => throw null;
    public override int GetHashCode() => 0;
    public override bool Equals(object other) => throw null;
}
```

Continuing to have this warning was a deliberate decision as the intent is to re-enforce that the base `Equals` is not supported
